platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolutionstrategy/DefaultResolutionStrategy.java

```diff
@@ -81,6 +81,7 @@ public class DefaultResolutionStrategy implements ResolutionStrategyInternal {
     private boolean verifyDependencies = true;
     private final Property<Boolean> useGlobalDependencySubstitutionRules;
     private boolean returnAllVariants = false;
+    private boolean keepStateRequiredForGraphResolution = true;
```

Comment:
The name and the flag are backwards now -- it should be default `false`, and the `if` below should invert it.